examples/prodlda.py

```diff
@@ -0,0 +1,207 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: ProdLDA
+================
+In this example, we will follow [1] to implement the ProdLDA topic model from
+Autoencoding Variational Inference For Topic Models by Akash Srivastava and Charles
+Sutton [2]. This model returns consistently better topics than vanilla LDA and trains
+much more quickly. Furthermore, it does not require a custom inference algorithm that
+relies on complex mathematical derivations. This example also serves as an
+introduction to Flax and Haiku modules in NumPyro.
+
+**References:**
+    1. http://pyro.ai/examples/prodlda.html
+    2. Akash Srivastava, & Charles Sutton. (2017). Autoencoding Variational Inference
+       For Topic Models.
+
+.. image:: ../_static/img/examples/prodlda.png
+    :align: center
+"""
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn.datasets import fetch_20newsgroups
+from sklearn.feature_extraction.text import CountVectorizer
+from wordcloud import WordCloud
+
+import haiku as hk
+import jax
+from jax import device_put, random
+import jax.numpy as jnp
+
+import numpyro
+from numpyro.contrib.module import haiku_module
+import numpyro.distributions as dist
+from numpyro.infer import SVI, TraceMeanField_ELBO
+
+
+class HaikuEncoder:
+    def __init__(self, vocab_size, num_topics, hidden, dropout_rate):
+        self._dropout_rate = dropout_rate
+        self._vocab_size = vocab_size
+        self._hidden = hidden
+        self._num_topics = num_topics
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+
+        h = jax.nn.softplus(hk.Linear(self._hidden)(inputs))
+        h = jax.nn.softplus(hk.Linear(self._hidden)(h))
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, h)
+
+        # NB: here we set `create_scale=False` and `create_offset=False` to reduce
+        # the number of learning parameters
+        h_mu = hk.Linear(self._num_topics)(h)
+        logtheta_loc = hk.BatchNorm(
+            create_scale=False, create_offset=False, decay_rate=0.1
+        )(h_mu, is_training)
+
+        h_sigma = hk.Linear(self._num_topics)(h)
+        logtheta_logvar = hk.BatchNorm(
+            create_scale=False, create_offset=False, decay_rate=0.1
+        )(h_sigma, is_training)
+        logtheta_scale = jnp.exp(0.5 * logtheta_logvar)
+        return logtheta_loc, logtheta_scale
+
+
+class HaikuDecoder:
+    def __init__(self, vocab_size, dropout_rate):
+        self._dropout_rate = dropout_rate
+        self._vocab_size = vocab_size
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, inputs)
+        h = hk.Linear(self._vocab_size)(h)
+        h = hk.BatchNorm(create_scale=True, create_offset=True, decay_rate=0.1)(
```

Comment:
You can set `create_scale=False` and `create_offset=False`

Reply:
Thanks! I suspected I had got something wrong with my batchnorms >_<